docs/llama_multi_lora_tutorial.md

```diff
@@ -0,0 +1,242 @@
++ You should expand the storage of these weight files to ensure they are logically organized in the workspace.
+
+## Step 4: Prepare `model repository` for Triton Server
+
+__4.1 Download the model repository files__
+
+To use Triton, a model repository is needed, for *model path* , *backend configuration* and other information. The vllm backend is implemented based on python backend, and `sampling_params` of vllm are sampled from `model.json`.
+
+To create a triton model repository, you may download the files through these commands:
+
+```bash
+# NOTICE: you must first cd to your vllm_workspace path.
+cd /vllm_workspace
+
+mkdir -p model_repository/vllm_model/1
+wget -P model_repository/vllm_model/1 https://raw.githubusercontent.com/triton-inference-server/vllm_backend/r<xx.yy>/samples/model_repository/vllm_model/1/model.json
+wget -P model_repository/vllm_model/ https://raw.githubusercontent.com/triton-inference-server/vllm_backend/r<xx.yy>/samples/model_repository/vllm_model/config.pbtxt
+```
+
+The model repository should look like this:
+
+```
+model_repository/
+└── vllm_model
+    ├── 1
+    │   └── model.json
+    └── config.pbtxt
+```
+
+---
+
+Now, you have finished the basic deployment, and the file structure should look like this:
+
+```
+vllm_workspace
+├── weights
+│   ├── backbone
+│   │   └── llama-7b-hf
+│   └── loras
+│       ├── alpaca-lora-7b
+│       └── bactrian-x-llama-lora-7b
+│
+└── model_repository
+    └── vllm_model
+        ├── 1
+        │   └── model.json
+        └── config.pbtxt
+```
+
+__4.2 Populate `model.json`__
+
+For this tutorial we will use the following set of parameters, specified in the `model.json`.
+
+```json
+{
+    "model":"/vllm_workspace/weights/backbone/llama-7b-hf",
+    "disable_log_requests": "true",
+    "gpu_memory_utilization": 0.8,
+    "tensor_parallel_size": 2,
+    "block_size": 16,
+    "enforce_eager": "true",
+    "enable_lora": "true",
+    "max_lora_rank": 16
+}
+```
+
++ `model`: The path to your model repository
++ `disable_log_requests`: To show logs when launch vllm or not.
++ `gpu_memory_utilization`: The gpu memory allocated for the model weights and vllm *PagedAttention* kv cache manager.
++ `tensor_parallel_size`: The vllm now support the tensor paralism, so you can decide how many gpus you want to use for serving.
++ `block_size`: vLLM kv cache block size.
++ `enable_lora`: If you want to support vllm multi-lora, this should be configured and set `true`.
++ `max_lora_rank`: The maximum of LoRA rank of your lora adapter.
+
+The full set of parameters can be found [here](https://github.com/Yard1/vllm/blob/multi_lora/vllm/engine/arg_utils.py#L11).
+
+__4.3 Specify local lora path__
+
+vLLM v0.3.0 just supported the inference of **local lora weights applying**, which means that the vllm cannot pull any lora adapter from huggingface. So triton should know where the local lora weights are.
```

Comment:
Has this changed for vLLM 0.4.0? Triton has advanced the vLLM version to 0.4.0.post1. See here: https://github.com/triton-inference-server/server/blob/main/build.py#L79
I believe we should mention same vLLM version that we are shipping with Triton 24.05. 

@oandreeva-nv  Thoughts?

Reply:
Let's confirm the behavior in vLLM 0.4.0.post1.